packages/repository/src/__tests__/unit/decorator/model-and-relation.decorator.unit.ts

```diff
@@ -132,6 +132,41 @@ describe('model decorator', () => {
     recentOrders?: Order[];
   }
 
+  it('hides a property defined as hidden', () => {
+    @model()
+    class Client extends Entity {
+      @property()
+      name: string;
+      @property({hidden: true})
+      password: string;
+
+      constructor(data: Partial<Client>) {
+        super(data);
+      }
+    }
+    const client = new Client({
+      name: 'name',
+      password: 'password',
+    });
+    expect(Client.definition.settings.hiddenProperties).to.containEql(
+      'password',
+    );
+    expect(client.toJSON()).to.eql({
+      name: 'name',
+    });
+  });
+  it('Throw error if desing type is not provided', () => {
+    const createModel = () => {
+      @model()
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      class Client extends Entity {
+        @property()
+        id: undefined;
```

Comment:
Could there be a more practical way to write this test case?

Reply:
sorry i'm not sure what else i could use, i created #6522 related to this (only undefined, null and maybe emitDecoratorMetadata to false throws undefined)

Reply:
Alright, we'll have to use a single commit message so please squash the two into one. Ready to land.